faiss/impl/ResultCollectorFactory.h

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+#include <faiss/impl/ResultCollector.h>
+namespace faiss {
+
+/** ResultCollectorFactory to create a ResultCollector object */
+struct ResultCollectorFactory {
+    DefaultCollector default_collector;
+    const std::vector<int64_t>* id_map;
+
+    // Create a new ResultCollector object
+    virtual ResultCollector* newCollector() {
```

Comment:
Should methods be `snake_case` instead of `camelCase`?

Reply:
In C++, the convention for method names is typically camel case.